packages/@lwc/synthetic-shadow/src/polyfills/event/polyfill.ts

```diff
@@ -74,14 +78,24 @@ function patchedTargetGetter(this: Event): EventTarget | null {
 
 function patchedComposedPathValue(this: Event): EventTarget[] {
     const originalTarget = eventTargetGetter.call(this);
-    const originalCurrentTarget = eventCurrentTargetGetter.call(this);
 
     // Account for events with targets that are not instances of Node (e.g., when a readystatechange
     // handler is listening on an instance of XMLHttpRequest).
     if (!(originalTarget instanceof Node)) {
         return [];
     }
 
+    // If the event target is a host element which has a native shadow root instance attached to it,
+    // invoke the native composedPath() method.
```

Comment:
This is an incorrect assumption since a shadow root could be attached to almost any DOM element. We could just check for the existence of an attached shadow root but would need some extra logic to differentiate between a current target that is a synthetic shadow root and a host element. This is because LWC binds shadow root event listeners to the host element.

Need to think about this some more...